src/web/components/Core/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,83 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
+import clsx from 'clsx';
+import { useCallback, useState } from 'react';
+
+import './MultiSelectDropdown.scss';
+
+type SelectOption = {
+  id: number;
+  name: string;
+};
+
+type MultiSelectDropdownProps = {
+  title: string;
+  options: SelectOption[];
+  onSelectedChange: (selected: Set<number>) => void;
+};
+
+export function MultiSelectDropdown({
+  title,
+  options,
+  onSelectedChange,
+}: MultiSelectDropdownProps) {
+  const [selectedItems, setSelectedItems] = useState<Set<number>>(new Set());
+  const [open, setOpen] = useState<boolean>(false);
+
+  const selectedCount =
+    selectedItems.size && selectedItems.size < options.length ? selectedItems.size : 'All';
+
+  const onOptionToggle = useCallback(
+    (id: number) => {
+      const newCheckedItems = new Set(selectedItems);
+      if (newCheckedItems.has(id)) {
+        newCheckedItems.delete(id);
+      } else {
+        newCheckedItems.add(id);
+      }
+
+      setSelectedItems(newCheckedItems);
+      onSelectedChange(newCheckedItems);
+    },
+    [onSelectedChange, selectedItems]
+  );
+
+  const checkboxItem = useCallback(
+    (option: SelectOption) => {
+      const checked = selectedItems.has(option.id);
+      return (
+        <DropdownMenu.CheckboxItem
+          className='multi-select-dropdown-checkbox-item'
+          checked={selectedItems.has(option.id)}
+          onCheckedChange={() => onOptionToggle(option.id)}
+          onSelect={(e) => e.preventDefault()} // Prevent dropdown close
+        >
+          <button
+            type='button'
+            role='checkbox'
+            aria-checked={checked}
+            className={clsx({ uncheck: !checked })}
+          >
+            {checked && <FontAwesomeIcon icon='check' />}
+          </button>
+          {option.name}
+        </DropdownMenu.CheckboxItem>
+      );
+    },
+    [onOptionToggle, selectedItems]
+  );
+
+  return (
+    <div className='multi-select-dropdown'>
+      <DropdownMenu.Root open={open} onOpenChange={setOpen}>
+        <DropdownMenu.Trigger className='multi-select-dropdown-trigger'>
+          {title}: {selectedCount}
```

Comment:
I see you asked about `selectedCount` in Figma - I also had the same question: should we show the names of the selected options or count of how many. In any case we can merge like this and change later though.

Reply:
yea, I think list all of them will make this really long/ do you remember in this morning's discussion, are we want to remove the `selectedCount` or not?

Reply:
yeah I think we decided to remove `selectedCount`. As for displaying the type names, tbh I don't think it's a problem if it is really long, I think we have plenty of horizontal screen space. In any case, I think it would be a good idea to make some styling change when filters are applied, maybe something like go from white background colour and black text (no selection) to blue background colour and white text (with selection)?

Reply:
Thanks, Got both title and styling updated

Reply:
looks good! one nit: could we possibly add a space after the commas when listing the options? Perhaps `formatStringsWithSeparator` could help, though I know the implementation is a bit different here.

Reply:
Actually I think it makes sense to reuse that formatStringsWithSeparator! Updated